crates/sqlexec/src/planner/physical_plan/insert.rs

```diff
@@ -76,8 +82,42 @@ impl ExecutionPlan for InsertExec {
                     "required table provider, found remote reference to insert".to_string(),
                 )),
                 ProviderReference::Provider(provider) => {
-                    // TODO: Add background job to track storage for native tables.
-                    Self::do_insert(provider, this.source, context).await
+                    let is_native_table = provider.as_any().downcast_ref::<NativeTable>().is_some();
```

Comment:
I guess I have a couple of questions:
- what do we do for writing storage for other non-native tables?
- is this supposed to be "bytes written since (this merges and/or the beginning of a table?)"
- it looks like "we try and enqueue a job every time there's a write, and it noops if there's something in the queue waiting to run; when the job runs (every 5m) we record the total current size; if the service shuts down while the job is in the queue, it never runs."
- don't we want something like: "we send a timestamp and a storage size (at the given moment), maybe a query ID or sequence number?, to a <service> internally, which adds it to a batch, which is flushed periodically to a delta file (parquet compresses this well, adding a bunch of rows in one shot is more effective than adding one row at a time; the data is more useful in this form, and it's possible to have "graceful shutdown" include this one last write...)